Source/Core/Core/IOS/Network/KD/NetKDRequest.cpp

```diff
@@ -141,9 +141,32 @@ IPCCommandResult NetKDRequest::IOCtl(const IOCtlRequest& request)
     break;
 
   case IOCTL_NWC24_REQUEST_SHUTDOWN:
-    // if ya set the IOS version to a very high value this happens ...
-    INFO_LOG(IOS_WC24, "NET_KD_REQ: IOCTL_NWC24_REQUEST_SHUTDOWN - NI");
+  {
+    if (request.buffer_in == 0 || (request.buffer_in & 3) != 0 || request.buffer_in_size < 8 ||
+        request.buffer_out == 0 || (request.buffer_out & 3) != 0 || request.buffer_out_size < 4)
+    {
+      return_value = IPC_EINVAL;
+      ERROR_LOG(IOS_WC24, "NET_KD_REQ: IOCTL_NWC24_REQUEST_SHUTDOWN = IPC_EINVAL");
+      break;
+    }
+
+    INFO_LOG(IOS_WC24, "NET_KD_REQ: IOCTL_NWC24_REQUEST_SHUTDOWN");
+    u32 event = Memory::Read_U32(request.buffer_in);
+    // TODO: Advertise shutdown event
+    if (event == 3)
```

Comment:
You may wish to omit this and instead expand the comment that the LEDs should be shut down if `event == 3`. Static analysis tools will flag empty conditional bodies like these as false positives otherwise.